### DISCUSSION TITLE
Do I need dropconnect at test time

### DISCUSSION THREAD

**Discussion by ElisonSherton** [baal-org/baal#171] [Q&A]
Hi guys,

If I have trained a `resnet50` model for 4 class multi-class classification problem incrementally using baal and if I wish to use this model in production for doing inference, then:

Is there a way to extract the above model and then load in production and predict without dropconnect i.e. deterministically all weights should be considered at the test time.

If not, then what is an ideal number of `iterations` that should be chosen to get reliable predictions (Assuming I perform a uniform averaging of probabilities obtained over all these forward passes).

Thanks,
Vinayak.

**Comment by Dref360** [comment]
Hello,

Is there an issue when you save the state_dict and load it back in a resnet50 without MCDropconnect?
If that's the case, I have some code to revert MCDropout, but not DropConnect, I can work on that and add it to BaaL.

If you want to use Bayesian averaging at test time, we found that 20 is a good compromise between performance and speed.
